### PR TITLE
Update tool-cache for download retries

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: __tests__/authutil.test.ts [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-    Math.random()
-      .toString(36)
-      .substring(7)
-  ),
-  'temp'
-);
-
-const rcFile = path.join(tempDir, '.npmrc');
-
-process.env['GITHUB_REPOSITORY'] = 'OwnerName/repo';
```
Comment (ericsciple) [inline __tests__/authutil.test.ts]: moved this code that sets env vars into beforeAll

File: __tests__/installer.test.ts [hunk 2]
```diff
@@ -98,6 +102,9 @@ describe('installer tests', () => {
   });
 
   it('Doesnt use version of node that was only partially installed in cache', async () => {
+    jest.spyOn(tc, 'downloadTool').mockImplementation(() => {
```
Comment (ericsciple) [inline __tests__/installer.test.ts]: remove this line

File: dist/index.js [hunk 3]
```diff
@@ -10096,9 +10096,12 @@ const os = __importStar(__webpack_require__(87));
 const path = __importStar(__webpack_require__(622));
 const httpm = __importStar(__webpack_require__(539));
 const semver = __importStar(__webpack_require__(280));
+const stream = __importStar(__webpack_require__(794));
```
Comment (ericsciple) [inline dist/index.js]: skip reviewing this file, this is the dist

File: __tests__/installer.test.ts [hunk 4]
```diff
@@ -1,54 +1,58 @@
...
-  'runner',
-  path.join(
-    Math.random()
-      .toString(36)
-      .substring(7)
-  ),
-  'temp'
-);
-
-process.env['RUNNER_TOOL_CACHE'] = toolDir;
```
Comment (ericsciple) [inline __tests__/installer.test.ts]: moved to beforeAll

File: __tests__/installer.test.ts [hunk 5]
```diff
@@ -1,54 +1,58 @@
...
-process.env['RUNNER_TEMP'] = tempDir;
+import * as io from '@actions/io';
+import * as tc from '@actions/tool-cache';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
 import * as installer from '../src/installer';
 
-const IS_WINDOWS = process.platform === 'win32';
+const isWindows = process.platform === 'win32';
```
Comment (ericsciple) [inline __tests__/installer.test.ts]: this variable had inconsistent naming from the style used everywhere else in this action :)

File: src/installer.ts [hunk 6]
```diff
@@ -1,5 +1,3 @@
-// Load tempDirectory before it gets wiped by tool-cache
-let tempDirectory = process.env['RUNNER_TEMPDIRECTORY'] || '';
```
Comment (ericsciple) [inline src/installer.ts]: RUNNER_TEMPDIRECTORY isnt a thing :(

Comment (ericsciple) [reply src/installer.ts]: i tested adding the action twice, and the first downloads and the second reuses the cache as expected. So not sure what the "wiped" comment used to mean.

File: src/installer.ts [hunk 7]
```diff
@@ -206,9 +189,11 @@ async function acquireNodeFromFallbackLocation(
   version: string
 ): Promise<string> {
   // Create temporary folder to download in to
-  let tempDownloadFolder: string =
+  const tempDownloadFolder: string =
```
Comment (ericsciple) [inline src/installer.ts]: This looks like a gap in coverage. Adding an E2E test for this.

**Comment by theomart77** [comment]
> _No description provided._

**Pull request merged**